### DISCUSSION TITLE
Open files independently of CliFM - closing CliFM shouldn't close the application launched

### DISCUSSION THREAD

**Discussion by vorboyvo** [leo-arch/clifm#315] [Q&A] [answered]
Hello,

I was trying to figure out how to launch files (using Lira) such that if I close CliFM, the application, be it the PDF reader or be it the text editor which I've configured to open in a new terminal window, doesn't close.

I've looked into:
* Using an external opener - I like Lira's configuration though and I don't really like xdg-open.
* Configuring it per filetype - I want this globally though, so no matter the type of file I want to open or what I open it with, it should behave in this way.

Maybe there's something obvious I'm missing - feel free to point it out to me if that's the case. Otherwise, is this possible? if so how can I do it?

Thank you!

**Comment by leo-arch** [comment]
Hi @vorboyvo. Just remove your mimelist file (`~/.config/clifm/profile/default/mimelist.clifm`) and restart **Clifm**: the file will be created anew. It should work out of the box.

The [current mimelist file](https://github.com/leo-arch/clifm/blob/master/misc/mimelist.clifm) uses the `%x` flag by default for GUI opening applications. This means: the file is opened in the background and both STDERR and STDOUT are silenced.

Example entry:

```sh
X:.*/pdf$=mupdf %x
```

TLDR; You're using an old mimelist file.

**Comment by vorboyvo** [comment]
Hi, I installed CliFM yesterday and other than modifications I made yesterday, my mimelist looks the same - it has the %x flag for GUI opening things. [Here](https://clbin.com/EwMwh) is my mimelist. I'm using sway on NixOS with alacritty if it may matter at all. I'm using nixpkgs/nixos-unstable and CliFM is up to date (1.21) on there.

**Reply by leo-arch** [reply]
Ok. Please run `mm info` on the file you're trying to open and report the output. What opening application is actually executed?

I've tried on `sway` (with `foot` terminal, and `mupdf` as opening application) and everything works as expected: the new window survives after closing **clifm** and even the terminal it was running on.

**Reply by vorboyvo** [reply]
```
<0> $ mm info CH\ 4\ Encheres\ 3.pdf
Name: CH 4 Encheres 3.pdf
MIME type: application/pdf
Associated application: zathura %x [MIME]
```

Also happens with:
```
<0> $ mm info notes.typ
Name: notes.typ
MIME type: text/x-objective-c
Associated application: alacritty -e kak %x [MIME]
```

In both cases `zathura` and `kak` inside `alacritty` are opened as expected.

I also tried with `foot` and same behaviour occurred.

**Reply by vorboyvo** [reply]
Wait, no. This is bizarre - with `foot` alone, it works exactly as I described it. But with `foot` and `mupdf` in the path, it works as you described, even for files opened by other applications such as the textfile given.

**Reply by leo-arch** [reply]
Just for clarification. Does the opened window survives after closing **clifm**? Does it survives after the the terminal **clifm** was running on? I guess you expect it to survive both.

**Reply by vorboyvo** [reply]
Ah, it survives after closing clifm, and survives after closing clifm and then closing the terminal, but not the terminal window while running clifm.

**Reply by leo-arch** [reply]
I see. Let me give this a try. I guess this is because you're unexpectedly closing **clifm**.

**Reply by leo-arch** [reply]
Reproduced pressing <kbd>Mod+Shitf+Q</kbd>. I'm not sure there's something we can do, but I'll take a look at it.

**Reply by vorboyvo** [reply]
Additional info that may be helpful: even closing clifm through `q` if I have launched it directly (through the .desktop file, which has `Exec=clifm` with `Terminal=true`, or by running `alacritty -e clifm`) quits the launched application - not just `Mod+Shift+Q`.

**Reply by leo-arch** [reply]
This thing should be working now. Compile and run the latest development version:

```sh
mkdir build
cd build
git clone https://github.com/leo-arch/clifm --depth=1
cd clifm
sudo make install
clifm
```
Make sure you're running the latest version (`1.21.13`) using the `version` command.

**Reply by vorboyvo** [reply]
Will do once i figure out how to with nix!

**Reply by leo-arch** [reply, marked as answer]
What's exactly the problem with Nix? Doesn't it have `make` or `git`? Or maybe some dependencies issue? In this latter case you can take a look at the [build dependencies](https://github.com/leo-arch/clifm/wiki/Introduction#1-satisfy-build-dependencies) section of the Wiki.

In any case, just clone the repo and run `make` and then `./clifm` to run the local copy generated by **make**(1).

**Reply by vorboyvo** [reply]
seems to work, thank you!

since your build dependencies section doesn't have an entry for nix, here's the build dependencies [from the package](https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/applications/file-managers/clifm/default.nix):
`libcap acl file readline python3`

**Reply by leo-arch** [reply]
You're welcome! And thanks for the dependencies list.

**Reply by vorboyvo** [reply]
Seems like the command 'x' still has this issue - with foot as well as with alacritty.

**Reply by leo-arch** [reply]
Should be working now.

**Reply by vorboyvo** [reply]
Yes!
